MovieApp/MovieWithSummaryCell.swift

```diff
@@ -0,0 +1,117 @@
+        self.movieId = movieId
+        self.router = router
+        Task {
+            await loadImage(imageURL: imageUrl, imageView: movieImage)
+        }
+        summaryLabel.text = summary
+    }
+    
+    private func assignViews() {
+        cellButton = UIButton()
+        cellButton.addTarget(self, action: #selector(handleCellButton), for: .touchUpInside)
+        contentView.addSubview(cellButton)
+        
+        movieNameLabel = UILabel()
+        cellButton.addSubview(movieNameLabel)
+        
+        movieImage = UIImageView()
+        cellButton.addSubview(movieImage)
+        
+        summaryLabel = UILabel()
+        cellButton.addSubview(summaryLabel)
+        
+    }
+    
+    private func style() {
+        backgroundColor = .white
+        contentView.layer.cornerRadius = 10
+        contentView.clipsToBounds = true
+        layer.shadowColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.1).cgColor
+        layer.shadowOpacity = 1
+        layer.shadowRadius = 20
+        layer.cornerRadius = 10
+        layer.shadowOffset = CGSize(width: 0, height: 4)
+        
+        movieImage.contentMode = .scaleAspectFill
+                
+        movieNameLabel.font = UIFont.boldSystemFont(ofSize: 16)
+        movieNameLabel.clipsToBounds = true
+        movieNameLabel.lineBreakMode = .byTruncatingTail
+        
+        summaryLabel.font = UIFont.systemFont(ofSize: 14)
+        summaryLabel.setContentHuggingPriority(UILayoutPriority.defaultHigh, for: .vertical)
+        summaryLabel.numberOfLines = 5
+        summaryLabel.lineBreakMode = .byTruncatingTail
+        summaryLabel.clipsToBounds = true
+        
+    }
+    
+    private func layout() {
+        movieImage.autoPinEdge(toSuperviewEdge: .top)
+        movieImage.autoPinEdge(toSuperviewEdge: .leading)
+        movieImage.autoPinEdge(toSuperviewEdge: .bottom)
+        movieImage.autoSetDimension(.width, toSize: 97)
+        
+        movieNameLabel.autoPinEdge(toSuperviewEdge: .top, withInset: 12)
+        movieNameLabel.autoPinEdge(.leading, to: .trailing, of: movieImage, withOffset: 16)
+        movieNameLabel.autoSetDimension(.height, toSize: 20)
+        movieNameLabel.autoPinEdge(toSuperviewEdge: .trailing)
+        
+        summaryLabel.autoPinEdge(.top, to: .bottom, of: movieNameLabel, withOffset: 8)
+        summaryLabel.autoPinEdge(toSuperviewEdge: .trailing, withInset: 12)
+        summaryLabel.autoPinEdge(.leading, to: .trailing, of: movieImage, withOffset: 16)
+        
+        cellButton.autoPinEdgesToSuperviewEdges()
+    }
+    
+    func loadImage (imageURL: String, imageView: UIImageView) async {
+        do {
+            let url = URL(string: imageURL)
+            let data = try Data(contentsOf: url!)
+            let image = UIImage(data: data)
+            imageView.image = image
+        } catch {
+            return
+        }
+    }
+    
+    @objc func handleCellButton() {
+        router.openMovieDetail(movieId: movieId)
+    }
```

Comment:
view should be in a sense "dumb", and not have a reference to the router at all, we should provide the view with any necessary **models**, but it should not be aware of any logic

try to achieve this with [didSelectItem](https://developer.apple.com/documentation/uikit/uicollectionviewdelegate/1618032-collectionview),[ delegate pattern](https://medium.com/@armanabkar/delegate-pattern-in-ios-a26f2b329097#:~:text=A%20design%20pattern%20that%20enables%20a%20class%20or%20structure%20to,an%20instance%20of%20another%20type&text=Delegation%20is%20a%20design%20pattern,an%20instance%20of%20another%20type.), or closure block